app/src/main/java/fr/gaulupeau/apps/Poche/tts/TtsService.java

```diff
@@ -443,37 +454,14 @@ private void playFakeSoundToMakeMediaSessionActive() {
     }
 
     private int requestAudioFocus() {
-        int audioFocusResult;
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
-            audioFocusResult = audioManager.requestAudioFocus(audioFocusChangeListener,
-                    AudioManager.STREAM_MUSIC, AudioManager.AUDIOFOCUS_GAIN);
-        } else {
-            if (audioFocusRequest == null) {
-                audioFocusRequest =
-                        new AudioFocusRequest.Builder(AudioManager.AUDIOFOCUS_GAIN)
-                                .setOnAudioFocusChangeListener(audioFocusChangeListener)
-                                .setWillPauseWhenDucked(true)
-                                .setAudioAttributes(
-                                        new AudioAttributes.Builder()
-                                                .setUsage(AudioAttributes.USAGE_MEDIA)
-                                                .setContentType(AudioAttributes.CONTENT_TYPE_SPEECH)
-                                                .build()
-                                )
-                                .build();
-            }
-            audioFocusResult = audioManager.requestAudioFocus(audioFocusRequest);
-        }
-        return audioFocusResult;
+        return AudioManagerCompat.requestAudioFocus(audioManager, audioFocusRequest);
     }
 
     private void abandonAudioFocus() {
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
-            audioManager.abandonAudioFocus(audioFocusChangeListener);
-        } else {
-            if (audioFocusRequest != null && audioManager.abandonAudioFocusRequest(audioFocusRequest)
-                    != AudioManager.AUDIOFOCUS_REQUEST_FAILED) {
-                audioFocusRequest = null;
-            }
+        if (audioFocusRequest != null
+                && AudioManagerCompat.abandonAudioFocusRequest(audioManager, audioFocusRequest)
+                != AudioManager.AUDIOFOCUS_REQUEST_FAILED) {
+            audioFocusRequest = null;
```

Comment:
Causes NPE in `requestAudioFocus()`.

Remove `audioFocusRequest != null` and `audioFocusRequest = null` here to always reuse one `audioFocusRequest`.

I guess, just `Log.d()` if `AudioManager.AUDIOFOCUS_REQUEST_FAILED`.

Reply:
Please add a `Log.d()` message if the result is `AudioManager.AUDIOFOCUS_REQUEST_FAILED`.
I have no confidence in how audiofocus is working, so I'd prefer to have some extra info just in case.

Reply:
Did I do it right?

Reply:
I meant `AudioManagerCompat.abandonAudioFocusRequest(audioManager, audioFocusRequest) == AudioManager.AUDIOFOCUS_REQUEST_FAILED`.